### PR TITLE
fix types & lint

### PR DESCRIPTION
**Pull request by keepingitneil**

### CODE REVIEW

File: livekit-agents/livekit/agents/voice_assistant/voice_assistant.py [hunk 1]
```diff
@@ -471,21 +478,18 @@ async def _play_speech(self, speech_info: _SpeechInfo) -> None:
...
                     continue
 
-                tool_calls.append(called_fnc.call_info)
                 tool_calls_results.append(
-                    llm.ChatMessage.create_tool_from_called_function(called_fnc)
+                    ChatMessage.create_tool_from_called_function(called_fnc)
                 )
 
             chat_ctx = speech_info.source.chat_ctx.copy()
-            chat_ctx.messages.extend(tool_calls)
```
Comment (keepingitneil) [inline livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:488]: @theomonnom Was this correct? Types were wrong

File: livekit-agents/livekit/agents/stt/stream_adapter.py [hunk 2]
```diff
@@ -24,7 +25,9 @@ def __init__(
         stt: STT,
         vad: VAD,
     ) -> None:
-        super().__init__(streaming_supported=True)
+        super().__init__(
+            capabilities=STTCapabilities(streaming=True, interim_results=False)
```
Comment (davidzhao) [inline livekit-agents/livekit/agents/stt/stream_adapter.py:29]: should this reflect the capabilities of the thing it's wrapping? i.e. copy(stt.capabilities.copy) but update `.streaming=True`

Comment (keepingitneil) [reply livekit-agents/livekit/agents/stt/stream_adapter.py:29]: yeah makes sense. interim_results will always be False in the thing it's wrapping (I think) but anyways I think it's cleaner to do the copy. Will fix

Comment (davidzhao) [reply livekit-agents/livekit/agents/stt/stream_adapter.py:29]: sg, yeah I guess it makes no difference today.. but might make a diff when more capabilities are added.

Comment (theomonnom) [reply livekit-agents/livekit/agents/stt/stream_adapter.py:29]: the StreamAdapter *upgrades* the capability of a specific STT. 
If you want to get the real capabilities of the STT, you can use the wrapped_stt property and retrieve capabilities

File: livekit-agents/livekit/agents/voice_assistant/voice_assistant.py [hunk 3]
```diff
@@ -533,8 +537,15 @@ def __init__(
...
-        self._last_final_transcript = user_msg.content
+    def on_new_synthesis(self, user_msg: ChatMessage) -> None:
+        str_content: str = ""
+        if isinstance(user_msg.content, str):
+            str_content = user_msg.content
+        elif isinstance(user_msg.content, list):
+            str_items = [item for item in user_msg.content if isinstance(item, str)]
+            str_content = " ".join(str_items)
+
+        self._last_final_transcript = str_content
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/voice_assistant/voice_assistant.py:549]: We can assume it is always str here, I removed the code

**Comment by theomonnom** [comment]
tests aren't passing for llm, let's do that in another PR. The reason is that the API changed

**Review by theomonnom — APPROVED** [review]
lgtm, thx
just fixed the stuff I just pushed

File: livekit-agents/livekit/agents/voice_assistant/voice_assistant.py (hunk 1 above)
Comment (theomonnom) [reply livekit-agents/livekit/agents/voice_assistant/voice_assistant.py]: ah, fixed it thx.

**Pull request merged**